corehq/apps/custom_data_fields/edit_model.py

```diff
@@ -19,6 +19,7 @@ class CustomDataFieldsForm(forms.Form):
     The main form for editing a custom data definition
     """
     data_fields = forms.CharField(widget=forms.HiddenInput)
+    purge_existing = forms.BooleanField(widget=forms.HiddenInput, required=False, initial=False)
```

Comment:
Oh man this code is all way too abstract.  I apologize on behalf of Ethan of 2 1/2 years ago.